packages/react-dom/src/client/ReactDOMFiberComponent.js

```diff
@@ -292,12 +291,11 @@ function setInitialDOMProperties(
         // textContent on a <textarea> will cause the placeholder to not
         // show within the <textarea> until it has been focused and blurred again.
         // https://github.com/facebook/react/issues/6731#issuecomment-254874553
-        const canSetTextContent = tag !== 'textarea' || nextProp !== '';
-        if (canSetTextContent) {
-          setTextContent(domElement, nextProp);
+        if (tag !== 'textarea' || nextProp !== '') {
+          domElement.textContent = nextProp;
         }
       } else if (typeof nextProp === 'number') {
-        setTextContent(domElement, '' + nextProp);
+        domElement.textContent = '' + nextProp;
```

Comment:
Why stringify here and not line 295?

Reply:
`textContent` applies stringify and sanitization AFAIK